cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/AbstractStorageProviderNamespaceAdmin.java

```diff
@@ -140,6 +140,14 @@ private void createLocation(NamespaceMeta namespaceMeta) throws IOException {
             "with sufficient privileges for the user and then try creating a namespace.",
           customNamespacedLocation.toString(), namespaceMeta.getNamespaceId()));
       }
+      // we also expect it to empty since non-empty directories can lead to various inconsistencies CDAP-6743
+      if (customLocation.list().length != 0) {
+        // TODO: Add username in the below exception message.
+        throw new IOException(String.format(
+          "The provided home directory '%s' for namespace '%s' is not empty. Namespace can only be mapped to an " +
+            "empty directory. Please try creating the namespace again with an empty directory mapping.",
```

Comment:
nit: the second sentence may not add much value. but feel free to ignore my comment.